cpp/src/umap/knn_graph/algo.cuh

```diff
@@ -27,6 +27,7 @@
 #include <raft/core/host_mdspan.hpp>
 #include <raft/core/mdspan.hpp>
 #include <raft/core/mdspan_types.hpp>
+#include <raft/distance/distance_types.hpp>
```

Comment:
Should we consider switching to cuVS? cc @jinsolp

Reply:
I think we should prioritize moving to the nn-descent implementation in cuvs (I'll open an issue). That's tangential to this PR though, which is about cleaning up some parameter handling. We need to use raft's distance type here since it's what the raft nn-descent requires.